src/Client.php

```diff
@@ -661,11 +762,23 @@ public function setAccessToken(string $accessToken): self
 
     /**
      * Get the HTTP headers.
+     *
+     * @param array $headers
+     * @return array
      */
     protected function getHeaders(array $headers = []): array
     {
-        return array_merge([
-            'Authorization' => "Bearer {$this->accessToken}",
-        ], $headers);
+        $auth = []; // defaults no-auth, can be used for public endpoints
+        if (isset($this->appKey, $this->appSecret)) {
+            $auth = [
+                'Authorization' => 'Basic '.base64_encode("{$this->appKey}:{$this->appSecret}"),
+            ];
+        } elseif (isset($this->accessToken)) {
```

Comment:
`else` is the bane of my existence. Refactor this in dedecated functions `getHeadersForCredentials` and `getHeadersForBearerToken`. Use a ternary for determining which of the functions you need.